github.cake

```diff
@@ -0,0 +1,78 @@
+#addin "nuget:https://api.nuget.org/v3/index.json?package=Octokit&version=0.48.0"
+#addin "nuget:https://api.nuget.org/v3/index.json?package=NuGet.Versioning&version=5.7.0"
+#addin "nuget:https://api.nuget.org/v3/index.json?package=Cake.Yaml&version=3.1.1"
+
+using Octokit;
+using NuGet.Versioning;
+
+static CakeGitHubReleaseInfo _cakeGitHubReleaseInfo;
+
+public static CakeGitHubReleaseInfo GetCakeGitHubReleaseInfo(this ICakeContext context, DirectoryPath releaseDir)
+{
+    if (!(_cakeGitHubReleaseInfo is null))
+    {
+        return _cakeGitHubReleaseInfo;
+    }
+
+    // We cache the latest Cake release information in a file to speed up local development and also
+    // to decrease the chance of getting rate limited by GitHub when running unauthenticated
+    var cachedGitHubDir = context.MakeAbsolute(releaseDir.Combine("cake-github"));
+    var cachedReleaseInfoFile = cachedGitHubDir.CombineWithFilePath($"{nameof(CakeGitHubReleaseInfo)}.yml");
+    if (context.FileExists(cachedReleaseInfoFile))
+    {
+        context.Verbose("Retrieving Cake release information from cached file {0}...", cachedReleaseInfoFile);
+
+        _cakeGitHubReleaseInfo = context.DeserializeYamlFromFile<CakeGitHubReleaseInfo>(cachedReleaseInfoFile);
+        LogGitHubReleaseInfo(context, _cakeGitHubReleaseInfo);
+
+        return _cakeGitHubReleaseInfo;
+    }
+
+    context.Information("Retrieving Cake release information from GitHub...");
+
+    var client = new GitHubClient(new ProductHeaderValue("cake-build-website"));
+
+    var gitHubAccessToken =  context.EnvironmentVariable("git_access_token");
+    if (!string.IsNullOrWhiteSpace(gitHubAccessToken))
+    {
+        client.Credentials = new Credentials(gitHubAccessToken);
+    }
+
+    var allCakeReleases = client.Repository.Release.GetAll("cake-build", "cake")
```

Comment:
Can we use `client.Repository.Release.GetLatest("cake-build", "cake").Result` here? I think this will ignore pre-release versions. But the question is if we want to show pre-release versions or not? For the banner on start page lates pre-release version probably makes most sense.

Reply:
Indeed `GetLatest` ignores prerelease versions (and thus returns v0.38.5 as of this writing) so it would not be an option if we want to show the latest prerelease in the banner (and link to the prerelease blog post).

The prerelease version is also the one where the source is downloaded from at the moment (even before this PR), thus all the API docs are using the prerelease assemblies, so I think it makes sense to promote the version that appears in the docs.

Reply:
Agree that it is correct for now. But I think I'll create an issue that we should think about which version we want to document in the API (IMHO for API documentation latest stable would make sense, while for the news banner on the home page I would also announce pre-release versions.